### PR TITLE
Fix registration

### PR DESCRIPTION
**Pull request by pathunstrom**
Changes GameEngine.register from accepting values to accepting callbacks. This allows dynamic values to be applied as part of event publication extension.

### CODE REVIEW

File: ppb/engine.py [hunk 1]
```diff
@@ -33,7 +35,7 @@ def __init__(self, first_scene: Type, *,
         # Engine State
         self.scenes = []
         self.events = deque()
-        self.event_extensions = defaultdict(dict)
+        self.event_extensions: DefaultDict[type, Dict[str, Callable[[], Any]]] = defaultdict(dict)
```
Comment (nbraud) [inline ppb/engine.py]: I don't think the type annotation is useful (it should be at the class level for `mypy` to be able to do something with it.)

Comment (pathunstrom) [reply ppb/engine.py]: I put it in because my linter was whining about types down in the publish function.

File: ppb/engine.py [hunk 2]
```diff
@@ -108,8 +110,9 @@ def publish(self):
         event = self.events.popleft()
         scene = self.current_scene
         event.scene = scene
-        for attr_name, attr_value in self.event_extensions[type(event)].items():
-            setattr(event, attr_name, attr_value)
+        extension_dict = self.event_extensions[type(event)]
+        for attr_name, attr_callback in extension_dict.items():
+            setattr(event, attr_name, attr_callback())
```
Comment (astronouth7303) [inline ppb/engine.py]: Shouldn't we pass it the event that's getting the attribute?

File: ppb/engine.py (hunk 1 above)
Comment (nbraud) [reply ppb/engine.py]: OK, that's weird.

**Comment by astronouth7303** [comment]
You haven't implemented the type wildcard?

**Comment by pathunstrom** [comment]
Right. Will get on that.

**Pull request merged**